File: T/167-two-sum-ii-input-array-is-sorted/main.go
package main

import "fmt"

func twoSumV2(nums []int, target int) []int {
	m := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		v, ok := m[target-nums[i]]
		if ok {
			return []int{v + 1, i + 1}
		} else {
			m[nums[i]] = i
		}
	}
	return []int{}
}

func twoSum(nums []int, target int) []int {
	s, e := 0, len(nums)-1
	for s < e {
		if nums[s]+nums[e] == target {
			return []int{s + 1, e + 1}
		} else if nums[s]+nums[e] < target {
			s++
		} else {
			e--
		}
	}
	return []int{}
}

func main() {
	fmt.Println("a")
}

File: T/167-two-sum-ii-input-array-is-sorted/main.rs
struct Solution {}

impl Solution {
  pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
    let mut start: usize = 0;
    let mut end: usize = numbers.len() - 1;
    while start < end {
      let sum = numbers[start] + numbers[end];
      if sum == target {
        break;
      } else if sum > target {
        end -= 1;
      } else {
        start += 1;
      }
    }
    vec![start as i32 + 1, end as i32 + 1]
  }
}
